File: genetic-packer/src/main/java/genetic/packer/mapper/ContainerToBoundsMapper.java
package genetic.packer.mapper;

import java.util.function.Function;

import genetic.packer.dto.request.ContainerDto;
import javafx.geometry.Bounds;
import javafx.geometry.ContainerBoundsBuilder;
import org.mapstruct.Mapper;

/**
 * @author piotr.larysz
 */
@Mapper
public abstract class ContainerToBoundsMapper implements Function<ContainerDto, Bounds> {

    @Override
    public Bounds apply(ContainerDto containerDto) {
        return new ContainerBoundsBuilder()
            .width(containerDto.getWidth())
            .height(containerDto.getHeight())
            .depth(containerDto.getDepth())
            .build();
    }
}

File: genetic-packer/src/main/java/genetic/packer/mapper/ContainerToBoundsMapper.kt
package genetic.packer.mapper

import genetic.packer.dto.request.ContainerDto
import genetic.packer.factory.ContainerBoundsFactory
import javafx.geometry.Bounds

abstract class ContainerToBoundsMapper : (ContainerDto) -> Bounds {

  override fun invoke(containerDto: ContainerDto) = ContainerBoundsFactory.createBounds(
      width = containerDto.width.toDouble(),
      height = containerDto.height.toDouble(),
      depth = containerDto.depth.toDouble()
  )
}